From d760294e5e209a57f13537076b452291af2da787 Mon Sep 17 00:00:00 2001
From: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Date: Fri, 23 Feb 2024 11:18:58 +0100
Subject: [PATCH 530/954] feat(stm32mp21): disable PWR_D3CR access on STM32MP21

PWR_D3CR is not present on STM32MP21.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 8456a7d2cf25de8dce59fffbced205188483006b feat(stm32mp21): disable PWR_D3CR access on STM32MP21]
Change-Id: Ib2ac874143e463df772c9120436a33c5bfc4edf7
Signed-off-by: Sebastien PASDELOUP <sebastien.pasdeloup-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/362446
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 812bcf8e16..95b1bf627f 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -702,7 +702,9 @@ static void __dead2 stm32_system_off(void)
 	/* Request standby2 */
 	mmio_write_32(pwr_base + PWR_CPU1CR, PWR_CPU1CR_PDDS_D1 | PWR_CPU1CR_PDDS_D2);
 	mmio_write_32(pwr_base + PWR_CPU2CR, PWR_CPU2CR_PDDS_D2);
+#if !STM32MP21
 	mmio_write_32(pwr_base + PWR_D3CR, PWR_D3CR_PDDS_D3);
+#endif /* !STM32MP21 */
 	stm32mp2_pll1_disable();
 
 	/* Do not maintain RETRAM memory content in Standby or Vbat */
-- 
2.53.0

